src/main/java/mainclass/exercises/streams/Library.java

```diff
@@ -0,0 +1,29 @@
+package mainclass.exercises.streams;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.OptionalDouble;
+
+public class Library {
+    public static void main(String[] args) {
+        List<FavoriteBook> books = new ArrayList<>();
+        books.add(new FavoriteBook("Watchmen", "Alan Moore", "graphic novel", 1987));
+        books.add(new FavoriteBook("Thinner", "Stephen King", "horror novel", 1984));
+        books.add(new FavoriteBook("The Stainless Steel Rat", "Harry Harrison", "comic science fiction novel", 1961));
+        System.out.println("The initial list of books: ");
+        books.forEach(System.out::println);
+        System.out.println("\nThe filtered one: ");
+        filterBooksByPublicationYear(books).stream().forEach(System.out::println);
+        System.out.println("\nThe average year of publication is: ");
+        calculateAveragePublicationYear(books).stream().forEach(System.out::println);
+    }
+
+    public static List<FavoriteBook> filterBooksByPublicationYear(List<FavoriteBook> book) {
+        return book.stream()
+                .filter(b -> b.year >= 1980).toList();
```

Comment:
The task is to "display books published after a certain year (e.g. 1950)".

Reply:
updated, thank you!